main/zoe/guide/README.md

```diff
@@ -64,176 +63,170 @@ handle the credit cards themselves.
-    }
-    if (!helpers.canTradeWith(keepHandle, tryHandle)) {
-      throw helpers.rejectOffer(tryHandle);
-    }
-    const keepAmounts = zcf.getCurrentAllocation(keepHandle);
-    const tryAmounts = zcf.getCurrentAllocation(tryHandle);
-    // reallocate by switching the amount
-    const handles =         harden([keepHandle, tryHandle]);
-    zcf.reallocate(handles, harden([tryAmounts, keepAmounts]));
-    zcf.complete(handles);
-    return defaultAcceptanceMsg;
-  }
+  const makeMatchingInvitation = firstSeat => {
+    assertProposalShape(firstSeat, {
+      give: { Asset: null },
+      want: { Price: null },
+    });
+    const { want, give } = firstSeat.getProposal();
 ```
-First, `swap` checks if the offer is still 
-active. If not, we reject the offer at hand. Second, if the offer 
-at hand isn't a match for the first offer, we want to reject it 
-for that reason as well.
 
-Once we're sure that we *do* have a matching offer, we can do the most
-exciting part, the reallocation.
+`makeMatchingInvitation()`, our first handler, then constructs a handler for the second offer,
+with the first offer's `want` and `give` in scope. This second
+handler, `matchingSeatOfferHandler()` does the final step.
+It uses the [`swap` helper  function](../api/zoe-helpers.html#swap-zcf-leftseat-rightseat-lefthasexitedmsg-righthasexitedmsg),
+a powerful Zoe Helper that handles a lot of the logic of doing a basic swap of assets.
 
+When the swap succeeds, it exits both seats and the contract shuts down. **tyg todo: Should we mention reallocation of assets between the parties?**
+```js
+    const matchingSeatOfferHandler = matchingSeat => {
+      const swapResult = swap(zcf, firstSeat, matchingSeat);
+      zcf.shutdown();
+      return swapResult;
+    };
+```
+Now let's put it together. The last step of the first handler, `makeMatchingInvitation()`
+is to create and return the second party's invitation, using
+`matchingSeatOfferHandler()` and including custom properties for the expected proposal.  **tyg todo: Maybe "offer proposal" to reinforce the connection?**
+```js
+    const matchingSeatInvitation = zcf.makeInvitation(
+      matchingSeatOfferHandler,
+      'matchOffer',
+      {
+        asset: give.Asset,
+        price: want.Price,
+      },
+    );
+    return matchingSeatInvitation;
+  };
+```
+Finally, we make the invitation for the first party, and return it as
+the contract's `creatorInvitation`.
+```js
+  const creatorInvitation = zcf.makeInvitation(
+    makeMatchingInvitation,
+    'firstOffer',
+  );
+  return { creatorInvitation };
+};
+```
+The `creatorInvitation` is only available to the contract instance's creator.
+(see [`startInstance`](../api/zoe.html#e-zoe-startinstance-installation-issuerkeywordrecord-terms)).
+The creator can use it (by making an offer with it) or send it to some other party.
+**tyg todo: I think the comment below is missing a close comment? Or does it just not recognize the
```

Comment:
I forgot to delete the other start comment. I was just adding this generally to the comment in case we want to later revivify it.